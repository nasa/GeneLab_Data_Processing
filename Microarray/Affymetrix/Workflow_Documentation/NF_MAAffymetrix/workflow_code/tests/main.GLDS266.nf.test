nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"

    test("GLDS-266: Should run without failures") {

        when {
            params {
                // define parameters here. Example: 
                osdAccession = "OSD-266"
                gldsAccession = "GLDS-266"
                biomart_attribute = "AFFY MoGene 2 1 st v1"
                limit_biomart_query = 300
            }
        }

        then {
            assert workflow.success
            assert snapshot(
                            path("${ params.outputDir }/Metadata").list(),
                            path("${ params.outputDir }/00-RawData").list(),
                            path("${ params.outputDir }/01-oligo_NormExp").list(),
                            path("${ params.outputDir }/02-limma_DGE").list(),
                            ).match()
        }

    }

}
