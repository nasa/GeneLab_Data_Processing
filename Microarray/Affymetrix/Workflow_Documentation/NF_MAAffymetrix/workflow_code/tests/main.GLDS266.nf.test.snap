{
    "GLDS-266: Should run without failures": {
        "content": [
            [
                "GLDS-266_metadata_GLDS-266-ISA.zip:md5,0ad945af82c091b89af4d5fe83c390e3",
                "OSD-266_microarray_v0_runsheet.csv:md5,2eb12f6b3ed2e2462a7181bcf129a7aa"
            ],
            [
                "raw_intensities_probe.csv:md5,36d94ef3076dfe8f5fda1d6dc0c5cbcf"
            ],
            [
                "normalized_expression_probeset.csv:md5,428bcd06838d037ab4f3eeba69251b10",
                "normalized_intensities_probe.csv:md5,a8cf4a61e248b4aea79b3e32f8e7a9f5"
            ],
            [
                "SampleTable.csv:md5,146814ee8f5207d870bcdd1696cabb26",
                "contrasts.csv:md5,72122612b7a36eed785979e2b95c5706",
                "differential_expression.csv:md5,5e66cfea992ac5e84a77d457b33f42b0",
                "visualization_PCA_table.csv:md5,b814788fea249b2548a9cc70ae1796b7"
            ]
        ],
        "timestamp": "2023-05-18T16:44:12+0000"
    }
}